### PR TITLE
Update extension to using TermInterface v2

### PR DESCRIPTION
**Pull request by ChrisRackauckas**

### CODE REVIEW

**Comment by jverzani** [comment]
Yes, I can tag. I had one other thing I wanted to close first, but that won't take any time.

But I'm seeing now that in SymPyPythonCall I have a SymbolicsExtension, which may be the reason behind all this. If so, I'd be happy to adopt the extension code from Symbolics into SymPyCore if that helps.

File: ext/SymPyCoreTermInterfaceExt.jl [hunk 1]
```diff
@@ -1,43 +1,43 @@
...
+Construct a new term with the operation f and arguments args, the term should be similar to t in type. if t is a TermInterface.Term object a new Term is created with the same symtype as t. If not, the result is computed as f(args...). Defining this method for your term type will reduce any performance loss in performing f(args...) (esp. the splatting, and redundant type computation). T is the symtype of the output term. You can use TermInterface.promote_symtype to infer this type. The exprhead keyword argument is useful when creating Exprs.
 ==#
-function SymbolicUtils.maketerm(t::SymPyCore.SymbolicObject, f, args, symtype=nothing;
+function TermInterface.maketerm(t::SymPyCore.SymbolicObject, f, args;
                                    metadata=nothing, exprhead=:call)
-    SymPyCore._similarterm(t, f, args, symtype; metadata=metadata, exprhead=exprhead)
+    SymPyCore._similarterm(t, f, args; metadata=metadata, exprhead=exprhead)
```
Comment (0x0f0f0f) [inline ext/SymPyCoreTermInterfaceExt.jl:40]: @ChrisRackauckas this is wrong it should be `maketerm(t, f, args, metadata)`

**Comment by 0x0f0f0f** [comment]
> Yes, I can tag. I had one other thing I wanted to close first, but that won't take any time.
> 
> But I'm seeing now that in SymPyPythonCall I have a SymbolicsExtension, which may be the reason behind all this. If so, I'd be happy to adopt the extension code from Symbolics into SymPyCore if that helps.

Wait before tagging!

File: ext/SymPyCoreTermInterfaceExt.jl (hunk 1 above)
Comment (0x0f0f0f) [reply ext/SymPyCoreTermInterfaceExt.jl:40]: https://juliasymbolics.github.io/TermInterface.jl/dev/#TermInterface.maketerm

Comment (0x0f0f0f) [reply ext/SymPyCoreTermInterfaceExt.jl:40]: t should also be a type

**Comment by jverzani** [comment]
@0x0f0f0f does https://github.com/jverzani/SymPyCore.jl/pull/72/files look correct?

**Comment by ChrisRackauckas** [comment]
> But I'm seeing now that in SymPyPythonCall I have a SymbolicsExtension, which may be the reason behind all this. If so, I'd be happy to adopt the extension code from Symbolics into SymPyCore if that helps.

yes that might be required to break the circular dep

**Pull request merged**